src/main/java/de/azapps/kafkabackup/common/partition/cloud/S3BatchWriter.java

```diff
@@ -0,0 +1,83 @@
+package de.azapps.kafkabackup.common.partition.cloud;
+
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import de.azapps.kafkabackup.common.record.Record;
+import de.azapps.kafkabackup.common.record.RecordJSONSerde;
+import de.azapps.kafkabackup.storage.s3.AwsS3Service;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.errors.RetriableException;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+public class S3BatchWriter {
+
+    private static final byte[] RECORD_SEPARATOR = {0x0A};  // utf-8 encoding of unix line feed
```

Comment:
instead of comment you can create variable with meaningful name like "UTF_8_UNIX_NEW_LINE"